keras/src/layers/preprocessing/image_preprocessing/mix_up.py

```diff
@@ -0,0 +1,163 @@
+from keras.src.api_export import keras_export
+from keras.src.layers.preprocessing.image_preprocessing.base_image_preprocessing_layer import (  # noqa: E501
+    BaseImagePreprocessingLayer,
+)
+from keras.src.random import SeedGenerator
+
+
+@keras_export("keras.layers.MixUp")
+class MixUp(BaseImagePreprocessingLayer):
+    """MixUp implements the MixUp data augmentation technique.
+
+    Args:
+        alpha: Float between 0 and 1. Inverse scale parameter for the gamma
+            distribution. This controls the shape of the distribution from which
+            the smoothing values are sampled. Defaults to 0.2, which is a
+            recommended value when training an imagenet1k classification model.
+        seed: Integer. Used to create a random seed.
+
+    References:
+        - [MixUp paper](https://arxiv.org/abs/1710.09412).
+        - [MixUp for Object Detection paper](https://arxiv.org/pdf/1902.04103).
+
+    Example:
+    ```python
+    (images, labels), _ = keras.datasets.cifar10.load_data()
+    images, labels = images[:10], labels[:10]
+    # Labels must be floating-point and one-hot encoded
+    labels = tf.cast(tf.one_hot(labels, 10), tf.float32)
+    mixup = keras.layers.preprocessing.MixUp(10)
+    augmented_images, updated_labels = mixup(
+        {'images': images, 'labels': labels}
+    )
+    # output == {'images': updated_images, 'labels': updated_labels}
+    ```
+    """
+
+    def __init__(self, alpha=0.2, seed=None, **kwargs):
```

Comment:
Please add support for the `data_format` argument (defaulting to None, and which could be `"channels_last"` or `"channels_first"`)

Reply:
I added data_format argument.